src/lib/containers/Login.tsx

```diff
@@ -47,104 +45,111 @@ class Login extends React.Component<Props, State> {
       hasLoginInFailed: false,
       isSignedIn: false,
       password: '',
-      username: ''
+      username: '',
     }
     this.handleChange = this.handleChange.bind(this)
     this.handleLogin = this.handleLogin.bind(this)
     this.getTokenView = this.getTokenView.bind(this)
     this.getLoginFailureView = this.getLoginFailureView.bind(this)
     this.onSignIn = this.onSignIn.bind(this)
   }
-    /**
-     * Updates internal state with the event that was triggered
-     *
-     * @param {*} event Form update
-     */
+  /**
+   * Updates internal state with the event that was triggered
+   *
+   * @param {*} event Form update
+   */
   public handleChange(event: React.FormEvent<HTMLInputElement>): void {
     const target = event.currentTarget
     const name = target.name
     const value = target.value
-    const newState: (Pick<any, any>) = { [name]: value }
+    const newState: Pick<any, any> = { [name]: value }
     this.setState(newState)
   }
-    /**
-     * Handle user login on click
-     *
-     * @param {*} clickEvent Userclick event
-     */
+  /**
+   * Handle user login on click
+   *
+   * @param {*} clickEvent Userclick event
+   */
   public handleLogin(clickEvent: React.FormEvent<HTMLElement>) {
     clickEvent.preventDefault() // avoid page refresh
-    const repoEndpoint = this.props.repoEndpoint ? this.props.repoEndpoint : SynapseClient.DEFAULT_ENDPOINT
-    const swcEndpoint = this.props.swcEndpoint ? this.props.swcEndpoint : SynapseClient.DEFAULT_SWC_ENDPOINT
-    SynapseClient.login(this.state.username, this.state.password, repoEndpoint)
-            .then((data: any) => {
-              SynapseClient.setSessionTokenCookie(data.sessionToken, swcEndpoint).then(() => {
-                // on session change, reload the page so that all components get the new token from the cookie
-                window.location.reload()
-              }).catch((errSetSession) => {
-                console.log('Could not set session token cookie!', errSetSession)
-                this.setState({
-                  errorMessage: errSetSession.reason,
-                  hasLoginInFailed: true,
-                  isSignedIn: false
-                })
-              })
-            })
-            .catch((err: any) => {
-              console.log('Handle login failed with err = ', err)
-              this.setState({
-                errorMessage: err.reason,
-                hasLoginInFailed: true,
-                isSignedIn: false
-              })
+    SynapseClient.login(this.state.username, this.state.password)
```

Comment:
Just a general thought, I wonder if it makes sense to start refactoring  nested promises into  async/awaits when we are in those areas doing other changes.

Reply:
I like that idea too! I can change this over to use async/await